crates/bevy_gltf/src/loader.rs

```diff
@@ -1981,46 +1988,160 @@ fn material_needs_tangents(material: &Material) -> bool {
 
 #[cfg(test)]
 mod test {
-    use std::path::PathBuf;
-
-    use super::resolve_node_hierarchy;
-    use crate::GltfNode;
-
-    impl GltfNode {
-        fn with_generated_name(index: usize) -> Self {
-            GltfNode {
-                index,
-                asset_label: crate::GltfAssetLabel::Node(index),
-                name: format!("l{}", index),
-                children: vec![],
-                mesh: None,
-                transform: bevy_transform::prelude::Transform::IDENTITY,
-                extras: None,
+    use std::path::Path;
+
+    use crate::{Gltf, GltfAssetLabel, GltfNode};
+    use bevy_app::App;
+    use bevy_asset::{
+        io::{
+            memory::{Dir, MemoryAssetReader},
+            AssetSource, AssetSourceId,
+        },
+        AssetApp, AssetPlugin, AssetServer, Assets, Handle, LoadState,
+    };
+    use bevy_core::TaskPoolPlugin;
+    use bevy_ecs::world::World;
+    use bevy_log::LogPlugin;
+    use bevy_render::texture::CompressedImageFormats;
+    use bevy_scene::ScenePlugin;
+    use bevy_utils::HashMap;
+
+    fn test_app(dir: Dir) -> App {
+        let mut app = App::new();
+        let reader = MemoryAssetReader { root: dir };
+        app.register_asset_source(
+            AssetSourceId::Default,
+            AssetSource::build().with_reader(move || Box::new(reader.clone())),
+        )
+        .add_plugins((
+            LogPlugin::default(),
+            TaskPoolPlugin::default(),
+            AssetPlugin::default(),
+            ScenePlugin::default(),
+        ))
+        // Somehow .finish() is not called on plugin, so we have to manually add stuff
```

Comment:
Maybe someone has an idea on how to fix it - without .finish() the actual loader isn't added.

Reply:
Found that I need to call finish/cleanup on app. Fixed now.